packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -42,8 +43,8 @@ export class MetadataGenerator {
     };
   }
 
-  private setProgramToDynamicControllersFiles(controllers: string[]) {
-    const allGlobFiles = importClassesFromDirectories(controllers);
+  private setProgramToDynamicControllersFiles(controllers: string[], esm: boolean) {
+    const allGlobFiles = importClassesFromDirectories(controllers, esm ? ['.mts', '.ts', '.cts']: ['.ts']);
```

Comment:
Does it hurt to add these in case we are CJS?

Reply:
I don't think it hurts. As far as I know you have 3 possible ways to work with ESM:
- keep `ts` files
- use `mts" files when you want to output specifically ESM files
- use `cts` files when you also want to output CommonJS

For our use case we only need `mts`/`mjs` files, but the logic is the same for the others as far as I know.